src/utils/config-defaults.js

```diff
@@ -0,0 +1,151 @@
+import { deepFreeze } from './object'
+
+// General BootstrapVue configuration
+//
+// BREAKPOINT DEFINITIONS
+//
+// Some components (BCol and BFormGroup) generate props based on breakpoints, and this
+// occurs when the component is first loaded (evaluated), which may happen before the
+// config is created/modified
+//
+// To get around this we make these components async (lazy evaluation)
+// The component definition is only called/executed when the first access to the
+// component is used (and cached on subsequent uses)
+//
+// See: https://vuejs.org/v2/guide/components-dynamic-async.html#Async-Components
+//
+// PROP DEFAULTS
+//
+// For default values on props, we use the default value factory function approach so
+// so that the default values are pulled in at each component instantiation
+//
+//  props: {
+//    variant: {
+//      type: String,
+//      default: () => getConfigComponent('BAlert', 'variant')
+//    }
+//  }
+
+// prettier-ignore
+export default deepFreeze({
```

Comment:
Do we really need to freeze the default config?
Wouldn't it be enough to make sure that we don't modify it anywhere?

Reply:
Just in case someone manually imports the default in thier app

As well... I want to remove the multiple calls to `cloneDeep` (cloning default, then cloning again afterwards) to improve installation performance.